tests/foreman/cli/test_flatpak.py

```diff
@@ -205,3 +205,112 @@ def test_scan_flatpak_remote(target_sat, function_org, function_product, remote)
+        7. Exercise flatpak actions against Satellite at the content host. (TBD: via REX)
+
+    :expectedresults:
+        1. Entire workflow works and allows user to install a flatpak app at the registered
+           contenthost.
+
+    """
+    sat, host = module_target_sat, module_flatpak_contenthost
+    # 1. Create a flatpak remote and scan it.
+    fr = sat.cli.FlatpakRemote().create(
+        {
+            'organization-id': function_org.id,
+            'url': FLATPAK_REMOTES['RedHat']['url'],
+            'name': gen_string('alpha'),
+            'username': settings.container_repo.registries.redhat.username,
+            'token': settings.container_repo.registries.redhat.password,
+        }
+    )
+    sat.cli.FlatpakRemote().scan({'id': fr['id']})
+    repos = sat.cli.FlatpakRemote().repository_list({'flatpak-remote-id': fr['id']})
+    assert len(repos), 'No repositories scanned'
+
+    # 2. Mirror a flatpak repository and sync it.
+    repo_names = ['rhel9/firefox-flatpak', 'rhel9/flatpak-runtime']  # runtime is dependency
+    remote_repos = [r for r in repos if r['name'] in repo_names]
+    for repo in remote_repos:
+        sat.cli.FlatpakRemote().repository_mirror(
+            {
+                'flatpak-remote-id': fr['id'],
+                'id': repo['id'],  # or by name
+                'product-id': function_product.id,
+            }
+        )
+        local_repo = sat.cli.Repository.list(
+            {'product-id': function_product.id, 'name': repo['name']}
+        )[0]
+        sat.cli.Repository.synchronize({'id': local_repo['id']})
+        assert 'latest' in sat.api.Repository(id=local_repo['id']).read().include_tags
+    local_repos = sat.cli.Repository.list({'product-id': function_product.id})
+    assert set([r['name'] for r in local_repos]) == set(repo_names)
+
+    # 3. Add the flatpak repo into a CV and publish+promote it into an LCE. (TBD)
+    # TBD; use Library/Defaut_org_view for now
+
+    # 4. Create an AK with the CVE and flatpak repo.
+    ak_lib = sat.cli.ActivationKey.create(
+        {
+            'name': gen_string('alpha'),
+            'organization-id': function_org.id,
+            'lifecycle-environment': 'Library',
+            'content-view': 'Default Organization View',
+        }
+    )
+    # ak_lib = {}
+    # ak_lib['name'] = 'AK1'
+
+    # 5. Register a content host using the AK.
+    res = host.register(function_org, None, ak_lib['name'], sat, force=True)
+    assert res.status == 0, (
+        f'Failed to register host: {host.hostname}\n' f'StdOut: {res.stdout}\nStdErr: {res.stderr}'
+    )
+    assert host.subscribed
+
+    # 6. Configure the contenthost to use Satellite's flatpak index. (TBD: via REX)
+    remote_name = f'SAT-remote-{gen_string("alpha")}'
+    res = host.execute(
+        'flatpak --user remote-add --authenticator-name=org.flatpak.Authenticator.Oci '
+        f'{remote_name} oci+https://{sat.hostname}/pulpcore_registry/'
```

Comment:
Is this the intended user flow or is this something that host registration should set up if it detects flatpak?

Reply:
It is the current workflow as per [this doc](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/administering_the_system_using_the_gnome_desktop_environment/assembly_installing-applications-using-flatpak_administering-the-system-using-the-gnome-desktop-environment#proc_enabling-the-red-hat-flatpak-remote_assembly_installing-applications-using-flatpak).

Anyway @sjha4 is working on a REX template to configure the host with a flatpak remote. I think the template _might_ be leveraged by the GR, but the GR update is rather part of the M2 epic if I'm not quite mistaken.

Reply:
Global registration will need to install podman and flatpak on the client and then run the remote-add. For now, we will leave it to the admins to set it up..